packages/commons/src/typeUtils.ts

```diff
@@ -0,0 +1,177 @@
+/**
+ * Returns true if the passed value is a record (object).
+ *
+ * @param value The value to check
+ */
+const isRecord = (value: unknown): value is Record<string, unknown> => {
+  return (
+    Object.prototype.toString.call(value) === '[object Object]' &&
+    !Object.is(value, null)
+  );
+};
+
+/**
+ * Check if a value is a string.
+ *
+ * @param value The value to check
+ */
+const isString = (value: unknown): value is string => {
+  return typeof value === 'string';
+};
+
+/**
+ * Check if a value is a number.
+ *
+ * @param value The value to check
+ */
+const isNumber = (value: unknown): value is number => {
+  return typeof value === 'number';
+};
+
+/**
+ * Check if a value is an integer number.
+ *
+ * @param value The value to check
+ */
+const isIntegerNumber = (value: unknown): value is number => {
+  return isNumber(value) && Number.isInteger(value);
+};
+
+/**
+ * Check if a value is truthy.
+ *
+ * @param value The value to check
+ */
+const isTruthy = (value: unknown): boolean => {
+  if (isString(value)) {
+    return value !== '';
+  } else if (isNumber(value)) {
+    return value !== 0;
+  } else if (typeof value === 'boolean') {
+    return value;
+  } else if (Array.isArray(value)) {
+    return value.length > 0;
+  } else if (isRecord(value)) {
+    return Object.keys(value).length > 0;
+  } else {
+    return false;
+  }
+};
+
+/**
+ * Check if a value is null.
+ *
+ * @param value The value to check
+ */
+const isNull = (value: unknown): value is null => {
+  return Object.is(value, null);
```

Comment:
what is the difference between `value === null` and `Object.is`? It looks like they yield same result, is it rather a preference?

Reply:
In the context of `null` the two notations are exactly equivalent, there are [other cases](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#description) where `Object.is` is preferable but this is not one of them.

In this case it was just preference. In my mind it _reads_ a bit more clear since it's closer to plain English.